### PR TITLE
Added more docs for bounce_margin_factor

### PR DESCRIPTION
**Pull request by solarkennedy**

### CODE REVIEW

File: docs/source/bouncing.rst [hunk 1]
```diff
@@ -152,10 +152,42 @@ Valid options are:
...
+
+The ``bounce_margin_factor`` setting in marathon yaml files controls how
+"aggressive" the bounce is in the face of failures. It only applies to the ``crossover`` bounce method.
+
+With the default setting (1.0) the ``crossover`` bounce will only begin to
+drain and kill old copies of the code once new copies are health to replace
+them. For example, if ``instances: 10`` and ``bounce_margin_factor: 1.0``
+(default), PaaSTA will not begin draining a single copy of the old 10 until at
+least one new copy of the service is healthy. If 10 new copies of the service
+are up, then it will being draining the old 10 copies right away. If the only
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: s/If the only/If only/

File: docs/source/bouncing.rst [hunk 2]
```diff
@@ -152,10 +152,42 @@ Valid options are:
 * ``check_haproxy``: Whether to check the local haproxy to make sure this task
   has been registered and discovered.
 
+Understanding How ``bounce_margin_factor`` Affects Bouncing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``bounce_margin_factor`` setting in marathon yaml files controls how
+"aggressive" the bounce is in the face of failures. It only applies to the ``crossover`` bounce method.
+
+With the default setting (1.0) the ``crossover`` bounce will only begin to
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: Please do not block-justify your RST source code like this - it turns small changes into large changes.
One sentence per line instead, please.

File: docs/source/bouncing.rst [hunk 3]
```diff
@@ -152,10 +152,42 @@ Valid options are:
...
+least one new copy of the service is healthy. If 10 new copies of the service
+are up, then it will being draining the old 10 copies right away. If the only
+one new copy of the service comes up, then the bounce will only drain one old
+copy as it gets replaced.
+
+The ``bounce_margin_factor`` adjusts how aggressive this procedure is. With the
+example of ``instances: 10`` and a ``bounce_margin_factor: 0.5`` (50%), then
+PaaSTA will preemptively being to drain and kill 5 copies of the old service to
+make room for the next 10 copies. The setting effectively gives PaaSTA
+"permission" to allow the service to dip below the set level of replication for
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: drop the scare quotes

Comment (EvanKrall) [reply docs/source/bouncing.rst]: s/for bounce purposes/during the bounce/

File: docs/source/bouncing.rst [hunk 4]
```diff
@@ -152,10 +152,42 @@ Valid options are:
...
+bounce purposes.
+
+The setting is most effective in situations where there are resource
+constraints. If the service is small and running in a large pool with plenty of
+headroom, a ``bounce_margin_factor`` is not necessary. A large service in a
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: s/A large service/For a large service/

File: docs/source/bouncing.rst [hunk 5]
```diff
@@ -152,10 +152,42 @@ Valid options are:
...
+small pool where there is no headroom, a ``bounce_margin_factor`` is essential,
+as we need to give PaaSTA permission to "make room" for the new copies of the service.
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: -scare quotes

File: docs/source/bouncing.rst [hunk 6]
```diff
@@ -33,10 +33,10 @@ predictable bouncing behavior is desired.
 Read more in the next section for exact details and differences amongst these
 bounce method.
 
-Note that in **no case** will PaaSTA "revert code or config" back to previous
-versions after a failed bounce. While some bounce methods have some protection
-against failure situations, PaaSTA will never undo the intent implied by a change.
-Addressing this problem should be done at at a layer *above PaaSTA*.
+Note that only in the case of ``paasta mark-for-deployoment --auto-rollback``
+will PaaSTA "revert code" back to previous versions after a failed
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: drop the scare quotes

File: docs/source/yelpsoa_configs.rst [hunk 7]
```diff
@@ -253,7 +253,8 @@ instance MAY have:
     to be drained when the crossover bounce method is used.
     0 < bounce_margin_factor <= 1. Defaults to 1 (no influence).
     This allows bounces to proceed in the face of a percentage of failures.
-    It doesn’t affect any other bounce method but crossover.
+    It doesn’t affect any other bounce method but crossover. See `the bounce docs <bouncing.html>`_
```
Comment (EvanKrall) [inline docs/source/yelpsoa_configs.rst]: One sentence per line.

Comment (EvanKrall) [reply docs/source/yelpsoa_configs.rst]: (This change is a good example of why I like the one-sentence-per-line rule. You're only adding text, but this shows up as a -/+)

Comment (solarkennedy) [reply docs/source/yelpsoa_configs.rst]: I've never heard of the one-sentence-per-line before

**Comment by solarkennedy** [comment]
Ok. I've made the lines longer and dropped quotes.

File: docs/source/bouncing.rst [hunk 8]
```diff
@@ -152,10 +152,33 @@ Valid options are:
...
+The ``bounce_margin_factor`` setting in marathon yaml files controls how aggressive the bounce is in the face of failures.
+It only applies to the ``crossover`` bounce method.
+
+With the default setting (1.0) the ``crossover`` bounce will only begin to drain and kill old copies of the code once new copies are health to replace them.
+For example, if ``instances: 10`` and ``bounce_margin_factor: 1.0`` (default), PaaSTA will not begin draining a single copy of the old 10 until at least one new copy of the service is healthy.
+If 10 new copies of the service are up, then it will being draining the old 10 copies right away.
+If only one new copy of the service comes up, then the bounce will only drain one old copy as it gets replaced.
+
+The ``bounce_margin_factor`` adjusts how aggressive this procedure is.
+With the example of ``instances: 10`` and a ``bounce_margin_factor: 0.5`` (50%), then PaaSTA will preemptively being to drain and kill 5 copies of the old service to make room for the next 10 copies.
```
Comment (EvanKrall) [inline docs/source/bouncing.rst]: s/being to drain and kill/begin by draining and killing/

**Pull request merged**